libcontainer/setns_init_linux.go

```diff
@@ -72,5 +70,14 @@ func (l *linuxSetnsInit) Init() error {
 			return newSystemErrorWithCause(err, "init seccomp")
 		}
 	}
-	return system.Execv(l.config.Args[0], l.config.Args[0:], os.Environ())
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	if err := label.SetProcessLabel(l.config.ProcessLabel); err != nil {
+		return err
+	}
+	err := system.Execv(l.config.Args[0], l.config.Args[0:], os.Environ())
+	// Ignore error, there is nothing we can do if this fails
+	label.SetProcessLabel("")
```

Comment:
I don't think that this line makes sense. Your commit message says "after the container execs", but `execv(2)` **does not return** unless there was some error when attempting to execute the process -- which means that unsetting the process label will only have an impact for the few milliseconds it takes to error out and exit.

If you really want to make this change, I would make it a `defer` to keep the code cleaner.

Reply:
If we can guarantee that no code will ever be execed after the system.Execv, then I am fine with removing the code.  Basically this line tells SELinux to go back to normal behaviour, IE DONT set the SELINUX Labels.  I was not sure if the Exec worked the same way in Go and the thread would not go back into a pool which go could use to launch other processes.

Reply:
> I was not sure if the Exec worked the same way in Go and the thread would not go back into a pool which go could use to launch other processes.

`system.Execv` calls an `execve(2)` under the hood which does everything that entails (kill all the other threads and execute the process).